### PR TITLE
patch for datetime with multiple languages

### LINKED ISSUE #826: Bug: date picker is not working when using another language than English
Date picker is not working for any of the translations and date is not shown properly.

Moreover, this bug prevents submitting QC results, preventing users from using QATrack+ at all, really.
When submitting, we get a message saying the date written in the date widget is not in a valid format.

I saw in the 4.0.0 release notes that localization is being advertised as being supported, but I'm thinking this might be premature, partly because of this issue (and also a lot of missing translations in v4.0.0).

And, because this is the Internet, I want to say I'm not being angry or reprimanding in any way :) I think it's great someone picked up the project and put so much work in it. I just think it might be best to say this is really supported in a later release.

French:

Image

English:

Image

I'm not sure how to fix this. It might be related to some JS library that qatrackplus is using to handle the date picker.

I saw this in v4.0.0 and develop.

### PR DESCRIPTION
**Pull request by crcrewso**
# Pull Request Template

## Summary

Introducing French format file so users of French installations can submit tests

## Related Issues

Patches #826. Proper fix to #826 requires db migrations. 

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature ( adds functionality, always assumed to be breaking until tested by maintainers)
- [ ] Breaking change (fix or addition that changes existing behaviour)
- [ ] Documentation only

## Target Branch

Pushing to develop, with cherry-picking into 4.0

## AI Usage Disclosure

- [ ] No AI tools were used in this PR.
- [x] AI tools were used:
  - **Tools / where used:** _e.g. GitHub Copilot for boilerplate_
  - [x] I've reviewed every AI-generated line as if I wrote it myself, and confirm: no PHI shared, no license-incompatible content (Apache-2.0).

---

### Coder Tasks

- [x] Rebased / merged `develop` branch; no merge conflicts.
- [x] `uv run pre-commit run --all-files` passes (ruff lint, ruff-format, django-upgrade).
  - `uv run ruff check . --fix` to help resolve linting errors
- [x] `uv run python -Wd manage.py check` reviewed for new deprecation warnings.
- [x] `uv run pytest` full test suite passes.
- [x] `uv run python manage.py makemigrations --check --dry-run` if migrations are found, change should be marked Breaking
- [ ] Docker builds succeed if touched.
- [ ] Docs updated under `docs/` if behaviour, settings, or APIs changed.
- [ ] `release_notes.md` updated if user-facing.
- [x] No secrets, debug code, or stray `print`/`console.log` left in.

### Review Code

- [ ] Reviewer verifies the target branch.
- [ ] For non-trivial changes: pulled the branch, ran it locally, and confirmed it works as described.
- [ ] CI is green; test suite and `pre-commit` pass.
- [ ] Reads through all changed files for anything weird or unintended.
- [ ] Logic is correct; edge cases and error handling considered.
- [ ] Migrations are sane and reversible where possible.
- [ ] Changes are consistent with the modernization direction (no reintroduction
      of patterns being migrated away from, e.g. flake8/yapf-era conventions).
- [ ] No security concerns (input validation, permissions, injection, XSS).
- [ ] Comments are sufficient and not overkill.
- [ ] Typos, spellcheck, professional.
- [ ] Reviewer sends back review.

### Edits

- [ ] Coder fixes any issues.
- [ ] Reviewer approves.

> Maintainer review is required for breaking changes, new features, and migrations.
> For bug fixes, refactors, and docs-only changes, reviewer approval is sufficient to merge.

---

## Testing Notes

Full test suite, manual testing, and careful review of the small number of code changes 

## Screenshots

N/A

### CODE REVIEW

File: qatrack/qa/views/forms.py [hunk 1]
```diff
@@ -365,7 +365,7 @@ def __init__(self, *args, **kwargs):
         for field in ('work_completed', 'work_started'):
             self.fields[field].widget = forms.widgets.DateTimeInput()
 
-            self.fields[field].widget.format = settings.DATETIME_INPUT_FORMATS[0]
+            self.fields[field].widget.format = settings.DATETIME_INPUT_FORMATS[2]
```
[the remark this answers is not in the document]
Comment (crcrewso) [reply qatrack/qa/views/forms.py:368]: This will be resolved in a future fix
